src/core/base_bitset_type.hpp

```diff
@@ -230,6 +230,32 @@ class BaseBitSet {
 		return (this->base() & Tmask) == this->base();
 	}
 
+	/**
+	 * Count the number of set bits.
+	 * @return The number of bits set to true.
+	 */
+	inline uint Count() const
+	{
+		return CountBits(this->base());
+	}
+
+	/**
+	 * Get the value of the Nth set bit.
+	 * @param n The Nth set bit from which we want to know the value.
+	 * @return The value of the Nth set bit, or std::nullopt if no Nth bit set.
+	 */
+	std::optional<Tvalue_type> GetNthSetBit(int n) const
+	{
+		if (n >= 0) {
```

Comment:
this feels like it should be an assert. why would you be asking for -1st set bit?